test/parallel/xyk-pallet.API.pool.deactivatedPool.test.ts

```diff
@@ -0,0 +1,254 @@
+  [testUser2] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token1, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser2)
+  );
+
+  await createPool(
+    testUser2.keyRingPair,
+    MGA_ASSET_ID,
+    defaultCurrencyValue,
+    token1,
+    defaultCurrencyValue
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("PoolAlreadyExists");
+  });
+});
+
+test("GIVEN deactivated pool WHEN another user tries to mint liquidity in the pool THEN user can do this", async () => {
+  [testUser2] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token1, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser2)
+  );
+
+  await mintLiquidity(
+    testUser2.keyRingPair,
+    MGA_ASSET_ID,
+    token1,
+    defaultCurrencyValue
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  const deactivatedPoolBalance = await getBalanceOfPool(MGA_ASSET_ID, token1);
+
+  expect(deactivatedPoolBalance[0][0]).bnGt(BN_ZERO);
+});
+
+test("GIVEN deactivated pool WHEN the user mints liquidity in the pool again THEN liquidity IDs are equal", async () => {
+  [testUser2] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token1, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser2)
+  );
+
+  await mintLiquidity(
+    testUser2.keyRingPair,
+    MGA_ASSET_ID,
+    token1,
+    defaultCurrencyValue
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  const liquidityIdAfter = await getLiquidityAssetId(MGA_ASSET_ID, token1);
+
+  expect(liquidityIdAfter).bnEqual(liquidityId);
+});
+
+test("GIVEN deactivated pool WHEN the user tries to swap/multiswap tokens on the deactivated pool THEN error returns", async () => {
+  let swapError: any = [];
+  try {
+    await sellAsset(
+      testUser1.keyRingPair,
+      MGA_ASSET_ID,
+      token1,
+      defaultCurrencyValue,
+      new BN(1)
+    );
+  } catch (error) {
+    swapError = error;
+  }
+  expect(swapError.data).toEqual(
+    "1010: Invalid Transaction: The swap prevalidation has failed"
+  );
+});
+
+test("GIVEN deactivated pool WHEN the user tries to compound reward on a deactivated pool THEN error returns", async () => {
+  await promotePool(sudo.keyRingPair, liquidityId, 20);
+
+  await compoundRewards(testUser1, liquidityId).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("MissingRewardsInfoError");
+  });
+});
```

Comment:
My bad, I mised that test sorry. 
But could you please move this test to the rewards test?  You can not compound rewards without a rewards setup.

Reply:
Perhaps you can simply promote the pool and validate the error.  Luckily it's about the 0,0 pool and not about the lack of rewards.